File: 387-first-unique-character-in-a-string/first-unique-character-in-a-string.cpp
class Solution {
public:
    int firstUniqChar(string s) {
        vector<int> alp(26,-2);
        for(int i=0;i<(int)s.size();++i){
            if(alp[s[i]-'a']==-2) alp[s[i]-'a']=i;
            else alp[s[i]-'a']=-1;
        }
        for(int i=0;i<(int)s.size();++i){
            if(alp[s[i]-'a']>=0) return i;
        }
        return -1;
    }
};

File: 387-first-unique-character-in-a-string/first-unique-character-in-a-string.java
class Solution {
    public int firstUniqChar(String s) {
        int n= s.length(), count;
        char ch;
        HashMap<Character,Integer> hm = new HashMap<>();
        for(int i=0;i<n;i++){
            ch= s.charAt(i);
            count= 0;
            if(hm.containsKey(ch)){
                count= hm.get(ch);
            }
            hm.put(ch,1+count);
        }
        for(int i=0;i<n;i++){
            if(hm.get(s.charAt(i))==1) return i;
        }
        return -1;
    }
}